controller/hybridgateway/builder/kongtarget_test.go

```diff
@@ -58,129 +57,95 @@ func TestKongTargetBuilder_WithLabels(t *testing.T) {
 		},
 		{
-			name: "backend ref with different namespace",
-			httpRoute: &gwtypes.HTTPRoute{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "test-route",
-					Namespace: "test-namespace",
-				},
-			},
-			backendRef: &gwtypes.HTTPBackendRef{
-				BackendRef: gatewayv1.BackendRef{
-					BackendObjectReference: gatewayv1.BackendObjectReference{
-						Name:      "test-service",
-						Namespace: &[]gatewayv1.Namespace{"other-namespace"}[0],
-						Port:      &port,
-					},
-					Weight: &weight,
-				},
-			},
-			expectedTarget: "test-service.other-namespace.svc.cluster.local:8080",
-			expectedWeight: 100,
+			name:           "IPv6 address with port",
+			host:           "2001:db8::1",
+			port:           8080,
+			expectedTarget: "[2001:db8::1]:8080",
 		},
 		{
-			name: "backend ref with empty namespace",
-			httpRoute: &gwtypes.HTTPRoute{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "test-route",
-					Namespace: "test-namespace",
-				},
-			},
-			backendRef: &gwtypes.HTTPBackendRef{
-				BackendRef: gatewayv1.BackendRef{
-					BackendObjectReference: gatewayv1.BackendObjectReference{
-						Name:      "test-service",
-						Namespace: &[]gatewayv1.Namespace{""}[0],
-						Port:      &port,
-					},
-					Weight: &weight,
-				},
-			},
-			expectedTarget: "test-service.test-namespace.svc.cluster.local:8080",
-			expectedWeight: 100,
+			name:           "hostname with port",
+			host:           "example.com",
+			port:           443,
+			expectedTarget: "example.com:443",
 		},
 		{
-			name: "backend ref with nil namespace",
-			httpRoute: &gwtypes.HTTPRoute{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "test-route",
-					Namespace: "test-namespace",
-				},
-			},
-			backendRef: &gwtypes.HTTPBackendRef{
-				BackendRef: gatewayv1.BackendRef{
-					BackendObjectReference: gatewayv1.BackendObjectReference{
-						Name:      "test-service",
-						Namespace: nil,
-						Port:      &port,
-					},
-					Weight: &weight,
-				},
-			},
-			expectedTarget: "test-service.test-namespace.svc.cluster.local:8080",
+			name:           "service FQDN with port",
+			host:           "my-service.default.svc.cluster.local",
+			port:           80,
+			expectedTarget: "my-service.default.svc.cluster.local:80",
+		},
+		{
+			name:           "localhost with custom port",
+			host:           "localhost",
+			port:           3000,
+			expectedTarget: "localhost:3000",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			builder := NewKongTarget().WithTarget(tt.host, tt.port)
+
+			target, err := builder.Build()
+			require.NoError(t, err)
+			assert.Equal(t, tt.expectedTarget, target.Spec.Target)
+		})
+	}
+}
+
+func TestKongTargetBuilder_WithWeight(t *testing.T) {
+	tests := []struct {
+		name           string
+		weight         *int32
+		expectedWeight int
+	}{
+		{
+			name:           "with weight 50",
+			weight:         func() *int32 { w := int32(50); return &w }(),
```

Comment:
the `github.com/samber/lo` package can be handy here
```suggestion
			weight:         lo.ToPtr(50)
```